### PR TITLE
Add Node.js release workflow

### PR DESCRIPTION
**Pull request by Thoralf-M**

### CODE REVIEW

File: .github/workflows/bindings-nodejs-publish.yml [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+      - name: Build project
+        run: yarn build
+
+      - name: Update Cargo.toml with git
+        run:  sed -i 's#path = "../core"#git = "https://github.com/iotaledger/iota-sdk", rev = "'$GITHUB_SHA'"#g' Cargo.toml 
+
+      - name: Print Cargo.toml
+        run: cat Cargo.toml
+
+      - name: Publish WASM bindings to NPM
```
Comment (thibault-martinez) [inline .github/workflows/bindings-nodejs-publish.yml]: Wasm?

File: .github/workflows/bindings-nodejs-publish.yml [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+      - name: Build Node.js prebuild (x64)
+        run: yarn run prebuild-x64
+        working-directory: bindings/nodejs
+
+      - name: Build Node.js prebuild (arm64)
+        run: yarn run prebuild-arm64
+        working-directory: bindings/nodejs
+
+      - name: Upload prebuild to GitHub release
+        run: npx prebuild --upload-all ${{ secrets.GITHUB_TOKEN }} --tag-prefix iota-sdk-nodejs-binding-v
```
Comment (thibault-martinez) [inline .github/workflows/bindings-nodejs-publish.yml]:
```suggestion
        run: npx prebuild --upload-all ${{ secrets.GITHUB_TOKEN }} --tag-prefix iota-sdk-nodejs-v
```
I'd just call it that

File: .github/workflows/bindings-nodejs-publish.yml [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        working-directory: bindings/nodejs
```
Comment (Alex6323) [inline .github/workflows/bindings-nodejs-publish.yml]: blank line at the end?

File: bindings/nodejs/scripts/download-node-lib-win-arm64.ps1 [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    $downloadedNodeVersion = $nodeVersion
+} Catch {
+    If ($_.Exception.Response -And $_.Exception.Response.StatusCode -eq "NotFound") {
+    Write-Output "No arm64 node.lib found for Node Windows $nodeVersion, trying fallback version $fallbackVersion..."
+    Invoke-WebRequest -Uri $fallbackUrl -OutFile $output
+    $downloadedNodeVersion = $fallbackVersion
+    }
+}
+
+Write-Output "Downloaded arm64 NodeJS lib v$downloadedNodeVersion to $output in $((Get-Date).Subtract($start_time).Seconds) second(s)"
```
Comment (Alex6323) [inline bindings/nodejs/scripts/download-node-lib-win-arm64.ps1]: here as well

File: .github/workflows/bindings-nodejs-publish.yml (hunk 3 above)
Comment (Thoralf-M) [reply .github/workflows/bindings-nodejs-publish.yml]: Added

File: bindings/nodejs/scripts/download-node-lib-win-arm64.ps1 (hunk 4 above)
Comment (Thoralf-M) [reply bindings/nodejs/scripts/download-node-lib-win-arm64.ps1]: Added

File: .github/workflows/bindings-nodejs-publish.yml [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+        run: npm i -g yarn
+
+      - name: Install JS dependencies
+        run: yarn
+
+      - name: Build project
+        run: yarn build
+
+      - name: Update Cargo.toml with git
+        run:  sed -i 's#path = "../core"#git = "https://github.com/iotaledger/iota-sdk", rev = "'$GITHUB_SHA'"#g' Cargo.toml
```
Comment (thibault-martinez) [inline .github/workflows/bindings-nodejs-publish.yml:44]: I get this locally 
```
bash-3.2$ sed -i 's#path = "../core"#git = "https://github.com/iotaledger/iota-sdk", rev = "'$GITHUB_SHA'"#g' Cargo.toml 
sed: 1: "Cargo.toml": invalid command code C
```

Comment (Thoralf-M) [reply .github/workflows/bindings-nodejs-publish.yml:44]: Maybe something macos specific? I tried it here https://github.com/Thoralf-M/iota-sdk/actions/runs/5311626540/jobs/9615082906 and there it worked and locally it works for me as well

Comment (thibault-martinez) [reply .github/workflows/bindings-nodejs-publish.yml:44]: Ok fair enough

**Review by thibault-martinez — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

**Pull request merged**